Adds a small api, along with api documentation.

Rdbaker opened the pull request.

mjperrone commented on README.rst:
```diff
@@ -78,3 +78,24 @@ This will generate a new migration script. Then run:
+then build the docs with middleman:
+::
+
+    bundle exec middleman build
```
There's nothing in `slate/`, so this doesn't work.

mjperrone replied:
oh slate is a submodule

mjperrone replied:
Something about the submodule wasn't set up properly? Or I'm supposed to do something on my end to deal with it being a submodule?

I haven't worked with submodules much, do you know what I'm supposed to do?

mjperrone replied:
OK following these instructions, in `skate/`, building the docs gives errors when building.

```
$ bundle exec middleman build
   identical  /Users/mjperrone/Development/GameCenter/gamecenter/static/docs/stylesheets/print.css
   identical  /Users/mjperrone/Development/GameCenter/gamecenter/static/docs/stylesheets/screen.css
   identical  ../gamecenter/static/docs/images/navbar.png
   identical  ../gamecenter/static/docs/images/logo.png
   identical  /Users/mjperrone/Development/GameCenter/gamecenter/static/docs/fonts/slate.svg
   identical  ../gamecenter/static/docs/fonts/slate.woff
   identical  ../gamecenter/static/docs/fonts/slate.woff2
   identical  ../gamecenter/static/docs/fonts/slate.ttf
   identical  ../gamecenter/static/docs/fonts/slate.eot
       error  ../gamecenter/static/docs/javascripts/all_nosearch.js
       error  ../gamecenter/static/docs/javascripts/all.js
      update  /Users/mjperrone/Development/GameCenter/gamecenter/static/docs/index.html
There were errors during this build, re-run with `middleman build --verbose` to see the full exception.
```

The javascript files aren't made/updated so the ToC and the tabbed code view don't work in the generated docs. Did you do anything different for setup than what you describe in this readme?

Rdbaker replied:
No, that's the exact setup I had. As long as you ran `bundle install` before running the build command

mjperrone commented on gamecenter/public/views.py:
```diff
@@ -5,12 +5,13 @@
+@blueprint.route("/")
 def home():
-    # Handle logging in
+    """Return the home page"""
```
I'm not a fan of these kinds of comments, but if I remember correctly, we've had this convo before so nbd leaving them.

Rdbaker replied:
I only put things like this in because my pep8 linter yells at me, and I'd rather it yell at me for more important things

mjperrone replied:
Your pep8 linter is inconsistently yelling at you? You haven't done this to all functions. Not saying you have to, just odd.

Rdbaker replied:
it yells at me for other things when it has bigger fish to fry

mjperrone commented on slate:
```diff
@@ -0,0 +1 @@
+Subproject commit c90eab79f9edbfb6292c24a43c0f52a7dbbb7554
```
Why as a submodule? Also slate docs say you should use a fork, but this commit exists in their repo, I'm confused.

Rdbaker commented:
"skate"

when I removed the .git/ dir from slate it wasn't enough to make the gamecenter repo recognize it as something that wasn't a submodule, so I just renamed it

mjperrone commented on gamecenter/api/views.py:
```diff
@@ -0,0 +1,76 @@
+
+
+def create_entry():
+    results, errors = score_schema.load(json.loads(request.data),
```
Haven't seen this style of error reporting since golang... weird

mjperrone commented on gamecenter/core/utils.py:
```diff
@@ -0,0 +1,18 @@
+"""A file for some utillity functions and classes."""
+
+
+class InvalidUsage(Exception):
```
does this belong in "utils" ?

Rdbaker replied:
maybe "core/exceptions.py" or "api/exceptions.py"?

mjperrone commented on gamecenter/settings.py:
```diff
@@ -5,9 +5,11 @@
 class Config(object):
-    SECRET_KEY = os_env.get('GAMECENTER_SECRET', 'secret-key')  # TODO: Change me
-    APP_DIR = os.path.abspath(os.path.dirname(__file__))  # This directory
+    # TODO: Change me
```
be more descriptive with this TODO, idk; what 'change me' means. Why change what, and how?

Rdbaker replied:
all the TODOs without a description were autogenerated, I'll go through and fix them

mjperrone commented on gamecenter/settings.py:
```diff
@@ -18,21 +20,23 @@ class ProdConfig(Config):
     DEBUG = False
-    SQLALCHEMY_DATABASE_URI = 'postgresql://localhost/example'  # TODO: Change me
-    DEBUG_TB_ENABLED = False  # Disable Debug toolbar
+    # TODO: Change me
```
same

mjperrone commented on skate/source/includes/_errors.md:
```diff
@@ -0,0 +1,14 @@
+
+Error Code | Meaning
+---------- | -------
+400 | Bad Request -- Your request is malformed
```
Are we going to use all these codes?

Rdbaker replied:
They were just some ideas for things we might use. We can even use 418 "I'm a teacup" if we want

mjperrone commented on skate/source/includes/_leaderboardget.md:
````diff
@@ -0,0 +1,187 @@
+```java
+import gamecenter
+
+gamecenter.setKey("myAPIKey");
````
We can talk about this when we actually make the java client, but I'd think you'd instantiate a gc object, initializing it with the key. That way if you have another gc object with another key, they won't conflict at all.

Rdbaker replied:
agreed, I wasn't thinking in Java-isms so much

mjperrone commented on skate/source/includes/_leaderboardget.md:
```diff
@@ -0,0 +1,187 @@
+
+
+
+## Get a user's scores
```
I'm going to OK these docs since I'm going to modify stuff in a minute.

mjperrone commented:
other than those small things, this looks great! Feel free to merge after you take a look at those comments
